Add "ICU on WebAssembly" notes to globalization-icu.md (#20830)
dotnet/docs: docs/standard/globalization-localization/globalization-icu.md

```
LD_SONAME = -Wl,-compatibility_version -Wl,$(SO_TARGET_VERSION_MAJOR) -Wl,-current_version -Wl,$(SO_TARGET_VERSION) -install_name @loader_path/$(notdir $(MIDDLE_SO_TARGET))
```

## ICU on WebAssembly

> In order to provide globalization compatibility with desktop profiles we aslo ship a special version of ICU for WebAssembly workloads. To make it as small as possible we had to introduce a couple of limitations, it allowed us to reduce ICU data file size from 24Mb down to 1.4Mb (~0.3Mb if compressed with Brotli). The following APIs are either not supported or supported with limitations:

Reviewer: ```suggestion
In order to provide globalization compatibility with desktop profiles we also ship a special version of ICU for WebAssembly workloads. To make it as small as possible we had to introduce a couple of limitations, that allowed us to reduce ICU data file size from 24Mb down to 1.4Mb (~0.3Mb if compressed with Brotli). The following APIs are either not supported or supported with limitations:
```

> - `CultureInfo.EnglishName`

Reviewer 2: ```suggestion
- <xref:System.Globalization.CultureInfo.EnglishName?displayProperty=nameWithType>
```

- `CultureInfo.NativeName`
- `DateTimeFormatInfo.NativeCalendarName`
- `RegionInfo.NativeName`

> - `String.Normalize` and `String.IsNormalized` don't support rarely used `NormalizationForm.FormKC` and `NormalizationForm.FormKD` forms.

Reviewer 2: ```suggestion

The following APIs are supported with limitations:

- <xref:System.String.Normalize(System.Text.NormalizationForm)?displayProperty=nameWithType> and <xref:System.String.IsNormalized(System.Text.NormalizationForm)?displayProperty=nameWithType> don't support the rarely used <xref:System.Text.NormalizationForm.FormKC?displayProperty=nameWithType> and <xref:System.Text.NormalizationForm.FormKD?displayProperty=nameWithType> forms.
```

> - `RegionInfo.CurrencyNativeName` returns the same value as `CurrencyEnglishName`.

Reviewer 2: ```suggestion
- <xref:System.Globalization.RegionInfo.CurrencyNativeName?displayProperty=nameWithType> returns the same value as <xref:System.Globalization.RegionInfo.CurrencyEnglishName?displayProperty=nameWithType>.
```
